### PR TITLE
Add support for Serbian Language Analyzer

### PR DESCRIPTION
**Pull request by velizarvel**
Starting from `org.apache.lucene.analysis` version 8.6, an Analyzer for the Serbian language is available. Within the `sr` folder, there is a `stopwords.txt` file that contains the default stopwords for the Serbian language. However, the current usage of Serbian stopwords does not function correctly because in the `Analysis `class, within the `NAMED_STOP_WORDS` map, an entry for Serbian has not been added.

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Review by cbuescher — CHANGES_REQUESTED** [review]
Hi @velizarvel thanks for your proposal to add the serbian stopword filter to Elasticsearch and apologies for the delay in replying. I'm wondering if we should also support for the Serbian analyzer that seems to also already exist in Lucene but that we don't seem to support yet. Did you consider using / adding this as well? I'm a bit hesitant to just add the stopword support because in itself it feels incomplete given there seems to be an analyzer that uses it.
In any case, we'd also need some basic tests added in "modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/analysis-common/20_analyzers.yml" and add the filter (or analyzer if this PR can be extended) to the documentation in the relevant place (see e.g. where we currently refer other stopword filters like "romanian" etc.)

**Comment by cbuescher** [comment]
@velizarvel thanks, I took a quick look and that already looks very promising. I will try and get a closer look hopefully soon. In the meantime, I saw you signed the CLA with a different email address than is used in a few of the latest commits in this PR (e.g. https://github.com/elastic/elasticsearch/commit/92412a08c2e451cb97b26eff8d3a12f061a64b25.patch uses a different mail). Since the CLA signing is tied to an email id, would you mind re-comminting those with the other address? Don't worry if you have to squash or force-push to the PR, I haven't started review and we will squash the whole PR in the end anyway.

**Comment by velizarvel** [comment]
Thank you @cbuescher. I've updated the commit author details to align with the email associated with the CLA. Everything should now be in order for your review. I apologize for the confusion caused by the different mail.

**Comment by cbuescher** [comment]
Thanks for adding the documentation and test, I will run our CI tests to see if we are missing anything.

@elasticmachine test this please

File: docs/reference/analysis/analyzers/lang-analyzer.asciidoc [hunk 1]
```diff
@@ -1588,6 +1589,55 @@ PUT /russian_example
...
+      }
+    }
+  }
+}
+----------------------------------------------------
+// TEST[s/"serbian_keywords",//]
+// TEST[s/\n$/\nstartyaml\n  - compare_analyzers: {index: serbian_example, first: serbian, second: rebuilt_serbian}\nendyaml\n/]
+
+<1> The default stopwords can be overridden with the `stopwords`
+or `stopwords_path` parameters.
```
Comment (cbuescher) [inline docs/reference/analysis/analyzers/lang-analyzer.asciidoc:1637]: I think we need anchors for these call-outs in L1606 and L1610 respectively

**Review by cbuescher — CHANGES_REQUESTED** [review]
There are some small changes needed for the docs tests to pass I think

**Comment by mark-vieira** [comment]
@elasticmachine test this please

**Comment by mark-vieira** [comment]
@elasticmachine ok to test

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci

File: docs/reference/analysis/analyzers/lang-analyzer.asciidoc (hunk 1 above)
Comment (velizarvel) [reply docs/reference/analysis/analyzers/lang-analyzer.asciidoc:1637]: Yes, you are right. I added the anchors in the required places in the new commit.

**Comment by cbuescher** [comment]
[buildkite] test this [please]

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
[buildkite] test this [please]

**Review by cbuescher — APPROVED** [review]

**Comment by cbuescher** [comment]
LGTM, I'm trying to get a CI run on this but our infra and tooling changed a bit recently. Please excuse the noise.

**Comment by cbuescher** [comment]
buildkite test this please

**Comment by cbuescher** [comment]
@velizarvel all green, many thanks for you work and adding support for this Lucene analyzer.

**Comment by velizarvel** [comment]
@cbuescher I'm glad I had the opportunity to contribute. Thank you for your cooperation.

**Pull request merged**